cosmo_tester/test_suites/snapshots/multi_tenant_snapshot_test.py

```diff
@@ -13,141 +13,182 @@
 #    * See the License for the specific language governing permissions and
 #    * limitations under the License.
 
-import json
-import os
-import uuid
-
 import pytest
 
-from cosmo_tester.framework.cluster import (
-    CloudifyCluster,
-    MANAGERS,
-)
-from cosmo_tester.framework.util import assert_snapshot_created
-
-from cloudify_cli.constants import DEFAULT_TENANT_NAME
-
 from . import (
-    deployment_id,
-    deploy_helloworld,
-    wait_for_execution,
-    update_client_tenant
+    assert_hello_worlds,
+    check_deployments,
+    check_from_source_plugin,
+    check_plugins,
+    _cluster,
+    confirm_manager_empty,
+    create_noinstall_helloworld,
+    create_snapshot,
+    delete_manager,
+    download_snapshot,
+    get_deployments_list,
+    get_secrets_list,
+    NOINSTALL_DEPLOYMENT_ID,
+    remove_and_check_deployments,
+    restore_snapshot,
+    set_client_tenant,
+    SNAPSHOT_ID,
+    upgrade_agents,
+    upload_and_install_helloworld,
+    upload_snapshot,
+    upload_test_plugin,
 )
 
-TENANTS = ['tenant_a', 'tenant_b', 'tenant_c']
-
-
-# TODO: Add version as master progresses
-@pytest.fixture(scope='module', params=['master'])
-def cluster(request, cfy, ssh_key, module_tmpdir, attributes, logger):
-    managers = (
-        MANAGERS[request.param](),
-        MANAGERS['master'](upload_plugins=False),
-    )
-
-    cluster = CloudifyCluster.create_image_based(
-            cfy,
-            ssh_key,
-            module_tmpdir,
-            attributes,
-            logger,
-            managers=managers,
-            )
 
+def test_restore_snapshot_and_agents_upgrade_multitenant(
+        cfy, cluster_multitenant, attributes, logger, tmpdir):
+    local_snapshot_path = str(tmpdir / 'snapshot.zip')
+    new_tenants = ('tenant1', 'tenant2')
+    hello_tenants = ('default_tenant', new_tenants[0])
+    noinstall_tenants = new_tenants
+
+    old_manager = cluster_multitenant.managers[0]
+    new_manager = cluster_multitenant.managers[1]
+    hello_vms = cluster_multitenant.managers[2:]
+
+    hello_vm_mappings = {
+        hello_tenants[0]: cluster_multitenant.managers[2],
```

Comment:
Again, this comes back to the fact that these VMs are managers. This is really awkward and confusing.